spec/services/representation_order_creator_spec.rb

```diff
@@ -94,4 +94,46 @@
       create
     end
   end
+
+  context 'with invalid defence_organisation data' do
+    let(:defence_organisation) do
+      {
+        organisation: {
+          address: {
+            postcode: 'Postcode Not Provided'
+          },
+          contact: {
+            home: 'Phone Not Provided',
+            work: 'Phone Not Provided',
+            mobile: 'Phone Not Provided',
+            fax: 'Phone Not Provided',
+            primary_email: 'Email Not Provided',
+            secondary_email: 'Email Not Provided'
+          }
+        }
+      }
+    end
+    let(:transformed_defence_organisation) do
+      {
+        organisation: {
+          address: {
+            postcode: nil
+          },
+          contact: {
+            home: nil,
```

Comment:
I think CP will not accept nil/null, and we might need to remove them before sending across.

Reply:
I'm now calling `delete` on any elements that fail validation, rather than setting them to `nil`.